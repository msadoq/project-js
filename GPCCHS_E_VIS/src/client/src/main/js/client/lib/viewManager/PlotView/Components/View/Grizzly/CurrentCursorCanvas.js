// ====================================================================
// HISTORY
// VERSION : 1.1.2 : DM : #3622 : 09/03/2017 : Moving DynamicView PlotView and TextView in
//  dataManager.
// VERSION : 1.1.2 : DM : #6835 : 31/05/2017 : First draft for parametric PlotView, x axis becomes
//  basic axis with numb values.
// VERSION : 1.1.2 : DM : #6835 : 28/08/2017 : PlotView is never parametric, fixed positionned
//  tooltip in Parametric and PlotView.
// VERSION : 1.1.2 : DM : #6835 : 08/09/2017 : Simplified style for canvas divs and tooltip divs,
//  calculated only once in main Chart component.
// END-HISTORY
// ====================================================================

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './GrizzlyChart.css';

export default class CurrentCursorCanvas extends PureComponent {

  static propTypes = {
    height: PropTypes.number.isRequired,
    width: PropTypes.number.isRequired,
    current: PropTypes.number.isRequired,
    xScale: PropTypes.func.isRequired,
    divStyle: PropTypes.shape().isRequired,
  }

  componentDidMount() {
    this.draw();
  }

  componentDidUpdate() {
    this.draw();
  }

  draw = () => {
    const {
      current,
      xScale,
      height,
      width,
    } = this.props;

    const ctx = this.el.getContext('2d');

    ctx.clearRect(0, 0, width, height);
    // =============== DRAWING
    ctx.beginPath();
    ctx.strokeStyle = '#11EE22';
    ctx.lineWidth = 2;
    ctx.moveTo(xScale(current), 0);
    ctx.lineTo(xScale(current), height);
    ctx.stroke();
    // ===============
  }

  assignEl = (el) => { this.el = el; }

  render() {
    const {
      height,
      width,
      divStyle,
    } = this.props;

    return (
      <canvas
        ref={this.assignEl}
        height={height}
        width={width}
        className={styles.canvas}
        style={divStyle}
      />
    );
  }
}
