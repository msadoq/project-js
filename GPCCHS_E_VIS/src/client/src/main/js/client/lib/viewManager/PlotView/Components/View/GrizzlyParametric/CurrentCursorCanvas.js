import React, { PropTypes, PureComponent } from 'react';
import styles from './GrizzlyChart.css';
import { divStyleType } from './types';

const { func, number } = PropTypes;

export default class CurrentCursorCanvas extends PureComponent {
  static propTypes = {
    divStyle: divStyleType.isRequired,
    current: number.isRequired,
    xScale: func.isRequired,
  };

  componentDidMount() {
    const {
      current,
      xScale,
      divStyle,
    } = this.props;
    const ctx = this.el.getContext('2d');
    draw(ctx, current, xScale, divStyle);
  }

  componentDidUpdate() {
    const {
      current,
      xScale,
      divStyle,
    } = this.props;
    const ctx = this.el.getContext('2d');
    draw(ctx, current, xScale, divStyle);
  }

  assignEl = (el) => { this.el = el; }

  render() {
    const {
      divStyle,
    } = this.props;

    return (
      <canvas
        ref={this.assignEl}
        height={divStyle.height}
        width={divStyle.width}
        className={styles.canvas}
        style={divStyle}
      />
    );
  }
}

/**
 * @param ctx
 * @param current
 * @param xScale
 * @param divStyle
 */
export const draw = (ctx, current, xScale, divStyle) => {
  const x = xScale(current);
  ctx.strokeStyle = '#1E2';
  ctx.fillStyle = '#1E2';
  ctx.lineWidth = 2;
  ctx.beginPath();
  ctx.clearRect(0, 0, divStyle.width, divStyle.height);
  ctx.moveTo(x, divStyle.height);
  ctx.lineTo(x, 0);
  ctx.stroke();
};
