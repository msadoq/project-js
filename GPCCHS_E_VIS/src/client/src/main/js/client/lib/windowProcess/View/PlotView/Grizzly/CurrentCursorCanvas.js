import React, { PropTypes, PureComponent } from 'react';
import styles from './GrizzlyChart.css';

export default class CurrentCursorCanvas extends PureComponent {

  static propTypes = {
    yAxesAt: PropTypes.string.isRequired,
    margin: PropTypes.number.isRequired,
    top: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    width: PropTypes.number.isRequired,
    current: PropTypes.number.isRequired,
    xScale: PropTypes.func.isRequired,
  }

  componentDidMount() {
    this.draw();
  }

  componentDidUpdate() {
    this.draw();
  }

  draw = () => {
    const {
      current,
      xScale,
      height,
      width,
    } = this.props;

    const ctx = this.el.getContext('2d');

    ctx.clearRect(0, 0, width, height);
    // =============== DRAWING
    ctx.beginPath();
    ctx.strokeStyle = '#11EE22';
    ctx.lineWidth = 2;
    ctx.moveTo(xScale(current), 0);
    ctx.lineTo(xScale(current), height);
    ctx.stroke();
    // ===============
  }

  assignEl = (el) => { this.el = el; }

  render() {
    const {
      height,
      width,
      yAxesAt,
      top,
      margin,
    } = this.props;

    const style = {};
    // horizontal position
    if (yAxesAt === 'left') {
      style.left = margin;
    } else if (yAxesAt === 'right') {
      style.right = margin;
    }

    // vertical position
    style.top = top;

    return (
      <canvas
        ref={this.assignEl}
        height={height}
        width={width}
        className={styles.canvas}
        style={style}
      />
    );
  }
}
