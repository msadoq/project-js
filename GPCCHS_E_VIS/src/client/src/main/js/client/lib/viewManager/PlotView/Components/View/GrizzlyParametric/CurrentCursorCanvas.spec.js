import { scaleLinear } from 'd3-scale';
import { draw } from './CurrentCursorCanvas';

const _ = require('lodash');

describe('CurrentCursorCanvas :: draw', () => {
  const propsStub = {
    current: 1509986030848,
    divStyle: {
      height: 1000,
      left: 0,
      top: 0,
      width: 1000,
    },
    height: 1000,
    width: 1000,
    xAxesAt: 'bottom',
    xScale: scaleLinear()
      .domain([0, 100])
      .range([0, 1000]),
  };
  const operations = [];
  const getContext = () => {
    const handler = {
      get: (target, property) => (...args) => operations.push(_.padEnd(property, 20) + args.join('|')),
      set: (target, property, value) => operations.push(_.padEnd(property, 20) + value),
    };
    return new Proxy({}, handler);
  };

  test('CurrentCursorCanvas :: draw :: nominal case', () => {
    const ctx = getContext();
    draw(
      ctx,
      propsStub.current,
      propsStub.xScale,
      propsStub.divStyle
    );
    expect(operations).toMatchSnapshot();
  });
});
