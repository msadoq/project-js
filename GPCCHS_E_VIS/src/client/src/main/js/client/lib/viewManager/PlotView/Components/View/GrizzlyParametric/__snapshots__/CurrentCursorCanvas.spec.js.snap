// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentCursorCanvas :: draw CurrentCursorCanvas :: draw :: nominal case 1`] = `
Array [
  "strokeStyle         #1E2",
  "fillStyle           #1E2",
  "lineWidth           2",
  "beginPath           ",
  "clearRect           0|0|1000|1000",
  "moveTo              15099860308480|1000",
  "lineTo              15099860308480|0",
  "stroke              ",
]
`;
